jenkinsbuilder/indexdindexymltransform.py

```diff
@@ -0,0 +1,196 @@
+from yaml import load, safe_dump
+from glob import glob
+import os
+from argparse import ArgumentParser
+
+
+class IndexGenerator:
+    """Thus class transforms index.d to index.yml and vice versa, Genearting one or the other, as needed"""
+
+    def __init__(self, indexd_path="./index.d", index_yml="./index.yml", reverse=False):
+        """Initialze a generator, giving path of index.d and index.yml and specifying is reverse transform needs to be
+        done"""
+
+        self._index_dotd = indexd_path
+        self._index_yml = index_yml
+        self._reverse = reverse
+
+        # If not reverse, check index.d folder for existence.
+        if not self._reverse:
+                if not os.path.exists(self._index_dotd):
+                    raise ValueError("Invalid path specified.")
+
+        # Otherwise, check for index.yml existance
+        else:
+            if not os.path.exists(self._index_yml):
+                raise ValueError("Invalid path specified")
+
+        return
+
+    def _genindexyml(self, potential_index_files):
+        """Generate index.yml from index.d files"""
+
+        current_index = 1
+        index_yml = {
+            "Projects": []
+        }
+
+        # Read every yaml file in list of yaml files
+        for potential_index_file in potential_index_files:
+
+            # Get the file basename by path split and take the 2nd part which is only the file name
+            file_name = os.path.split(potential_index_file)[1]
+            namespace = ""
+
+            # If a filename contains a ., take everything before the . as project namespace
+            if "." in file_name:
+                namespace = file_name.split(".")[0]
+
+            # Otherwise the project namespace is the entire filename
+            else:
+                namespace = file_name
+
+            potential_index_data = None
+
+            # Read the yaml data in the file
+            with open(potential_index_file, "r") as thefile:
+                potential_index_data = load(thefile)
+
+            # Go through every entry in the yaml data file, extracting what is needed.
+            for entry in potential_index_data:
+                tid = current_index
```

Comment:
why unnecessarily copy the dict data into vars, if you are not changing something? directly consume the values from the dict in line 73 onwards